### ISSUE TITLE
Office.context.mailbox.item.end.setAsync() set event time zone to UTC on Outlook Web

### ISSUE THREAD

**Issue by HongpengRen** [OfficeDev/office-js#2198]
# Office.context.mailbox.item.end.setAsync() set event time zone to UTC on Outlook Web

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: __Outlook Web____
* Host [Excel, Word, PowerPoint, etc.]: ___Outlook Web___
* Office version number: ______
* Operating System: ______
* Browser (if using Office on the web): ___Google Chrome___
 
## Expected behavior
Calling Office.context.mailbox.item.end.setAsync() changes appointment end time without reset event time zone to UTC.

## Current behavior
Calling Office.context.mailbox.item.end.setAsync() changes appointment end time and reset event time zone to UTC.
before:
![before](https://user-images.githubusercontent.com/13360654/138230533-6b17817d-a75f-4640-8cea-95098c526677.png)
after:
![after](https://user-images.githubusercontent.com/13360654/138230744-982c2076-7cb3-42f9-87ef-488361d92e2f.png)

## Steps to reproduce
1. Call Office.context.mailbox.item.end.setAsync() to change Outlook appointment event end time on Outlook Web.
2. Check the event time zone.

## Link to live example(s)
`// Link to full sample: https://raw.githubusercontent.com/OfficeDev/office-js-snippets/prod/samples/outlook/90-other-item-apis/get-set-end-appointment-organizer.yaml
Office.context.mailbox.item.start.getAsync((result) => {
  if (result.status !== Office.AsyncResultStatus.Succeeded) {
    console.error(`Get start date failed with message ${result.error.message}`);
    return;
  }

  var end = result.value; // Set end to current start date and time.
  end.setDate(end.getDate() + 1); // Set end as 1 day later than start date.
  Office.context.mailbox.item.end.setAsync(end, (result) => {
    if (result.status !== Office.AsyncResultStatus.Succeeded) {
      console.error(`Set end date failed with message ${result.error.message}`);
      return;
    }
    console.log(`Successfully set end date and time to ${end}`);
  });
});`

# Provide additional details

## Context
Trying to set appointment start and end time from an Outlook add-in, but on Outlook web, time zone is also reset to UTC.

## Useful logs
- [ ] Console errors
- [x] Screenshots
- [ ] Test file (if only happens on a particular file)

**Comment by isabela-dominguez** [comment]
Thank you for sharing this issue @HongpengRen! Connecting you with @exextoc who may be able to help.

**Comment by exextoc** [comment]
@HongpengRen Thanks for reporting this issue regarding `Time.setAsync()` api. It has been put on our backlog. We unfortunately have no timelines to share at this point. Internal Tracking Id 132301

**Comment by houfukude** [comment]
Hi there,
I also encountered the same problem.

In my office365 developer subscription, some accounts have problems. The other part doesn't have this problem.

Tried using `new Date()`
Tried using `moment().utc().format()`
Tried using `Office.context.mailbox.convertToLocalClientTime` and `Office.context.mailbox.convertToUtcClientTime`

But there are still problems.

Is there any update for this bug?
